### DISCUSSION TITLE
Unable to parse message: "PauseVideo" (unknown variant `PauseVideo`, expected one of `PointerEvent`, `WheelEvent`, `KeyboardEvent`, etc...

### DISCUSSION THREAD

**Discussion by KarimElsayad247** [H-M-H/Weylus#303] [Q&A] [answered]
So protocol.rs has this

```
pub enum MessageInbound {
    PointerEvent(PointerEvent),
    WheelEvent(WheelEvent),
    KeyboardEvent(KeyboardEvent),
    GetCapturableList,
    Config(ClientConfiguration),
    PauseVideo,
    ResumeVideo,
    ChooseCustomInputAreas,
}
```

and I was experimenting with these to get a feel for them, but sending PauseVideo resulted in an error

```
Unable to parse message: "PauseVideo" (unknown variant `PauseVideo`, expected one of `PointerEvent`, `WheelEvent`, `KeyboardEvent`, `TryGetFrame`, `GetCapturableList`, `Config` at line 1 column 12)
```

Are these unimplemented?

**Comment by H-M-H** [comment]
You are probably using an outdated version of Weylus. Try building it yourself or use a built from here: https://github.com/H-M-H/Weylus/actions/runs/11405365806

**Reply by KarimElsayad247** [reply]
I think I have the latest version, which was released on 31 of October
```
✗❯ weylus --version
weylus 0.11.4
```

And the portion of the code hasn't been changed since 37776a86dcb4bb8b334821686e69049a8a06e8bd which was a commit in 2020

**Reply by KarimElsayad247** [reply]
Could it be that this error occurs when there is no video stream running?

**Reply by KarimElsayad247** [reply]
Is there a way to enable verbose logging?

**Reply by KarimElsayad247** [reply]
Ok it's now suddenly working...? I don't understand tbh... like I just terminated the app to check flags, ran it again, and suddenly it's ok with it.

**Reply by H-M-H** [reply, marked as answer]
There has not been a release in a while and you are looking at https://github.com/H-M-H/Weylus/blob/00a19aa13910f645c6115870863be96d99d56389/src/protocol.rs#L15-L25 which is the master as of now. But v0.11.4 differs from current master: https://github.com/H-M-H/Weylus/blob/10a279b5ad203132b696f88b9596f91559e8e785/src/protocol.rs#L14-L25

**Reply by H-M-H** [reply]
Verbose logging can be enabled by setting `WEYLUS_LOG_LEVEL=TRACE` or `WEYLUS_LOG_LEVEL=DEBUG` as environment variables.

**Reply by KarimElsayad247** [reply]
Thanks

**Reply by KarimElsayad247** [reply]
BTW I just forgot to connect to the WS server that's why I thought the problem went away lol

**Reply by KarimElsayad247** [reply]
Oh.... latest release was on 31st October **2021**.... I should go to sleep.

**Reply by H-M-H** [reply]
Good night then  :)
